src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -273,7 +286,9 @@ - (void)finishRequestWithErrorHandler:(void (^)())errorHandler {
 }
 
 - (void)writesFinishedWithError:(NSError *)errorOrNil {
-  _requestWriter = nil;
+  @synchronized(_requestWriter) {
```

Comment:
I'm surprised Xcode's 'Analyze' command didn't flag this: _requestWriter is sometimes nil (in fact it is set to nil inside the block) and you should never @synchronize(nil)

did you mean:
@synchronized(self) { ... } ?